### DISCUSSION TITLE
General “best practices” question on using Prisma with await/async

### DISCUSSION THREAD

**Discussion by heymartinadams** [prisma/prisma#3154] [Q&A] [answered]
Noob question here. If I want to send data to the database, but don’t depend upon a response, can I skip using `await` with Prisma Client?

For example, if, in the client, we have a successful payment and I want to update the user, but don’t want to wait until I got confirmation from the backend that the user has been updated, can I do that? And even if I could, should I?

## Client-side
```js
useEffect(async () => {
	if (paymentIntent) {
			// OPTION #A:
			const user = await fetchApi('/api/user/update', {
				paymentId: paymentIntent.id
			})

			// OPTION #B:
			fetchApi('/api/user/update', {
				paymentId: paymentIntent.id
			})

			// Do something else, e.g. redirect to another page.
		}
	}, [paymentIntent])
```

## Backend
```js
const updateUser = async (req, res) => {
	const prisma = new PrismaClient()

	// OPTION #A
	const user = await prisma.user.update({...})
	return res.json({ user })

	// OPTION #B
	prisma.user.update({...})
	return res.json({ done: true })
}
```

**Comment by ryands17** [comment]
Hey @heymartinadams 👋 
I wouldn't recommend this as it wouldn't save you a lot of time, a better option would be to show an optimistic response and revert if the request has failed depending upon your use case.

**Comment by heymartinadams** [comment, marked as answer]
Thank you, @ryands17 🙏  What do you mean by “optimistic response”?

**Reply by ryands17** [reply]
By optimistic response, I mean upadting the data successfully on the frontend while the API is being executed and if the response is successful, keep it as it is and if it fails, revert to the original data. It is explained quite well in [this](https://uxdesign.cc/the-optimistic-ui-with-react-f1420e317d54) post.

**Reply by heymartinadams** [reply]
Got it, @ryands17, thank you!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
